crates/anyref-xform/src/lib.rs

```diff
@@ -0,0 +1,730 @@
+        }
+        let name = format!("closure{}", idx);
+        let f = self.function(&name, anyref, ret_anyref);
+        let ret = self.next_element;
+        self.next_element += 1;
+        self.elements.insert(ret, (idx, f));
+        ret
+    }
+
+    fn function(&self, name: &str, anyref: &[(usize, bool)], ret_anyref: bool) -> Function {
+        Function {
+            name: name.to_string(),
+            args: anyref.iter().cloned().collect(),
+            ret_anyref,
+        }
+    }
+
+    pub fn anyref_table_id(&self) -> TableId {
+        self.table.unwrap()
+    }
+
+    pub fn run(&mut self, module: &mut Module) -> Result<(), Error> {
+        if !self.enabled {
+            return Ok(());
+        }
+        let table = self.table.unwrap();
+
+        // Inject a stack pointer global which will be used for managing the
+        // stack on the anyref table.
+        let init = InitExpr::Value(Value::I32(DEFAULT_MIN as i32));
+        let stack_pointer = module.globals.add_local(ValType::I32, true, init);
+
+        let mut heap_alloc = None;
+        let mut heap_dealloc = None;
+
+        // Find exports of some intrinsics which we only need for a runtime
+        // implementation.
+        for export in module.exports.iter() {
+            let f = match export.item {
+                walrus::ExportItem::Function(f) => f,
+                _ => continue,
+            };
+            match export.name.as_str() {
+                "__wbindgen_anyref_table_alloc" => heap_alloc = Some(f),
+                "__wbindgen_anyref_table_dealloc" => heap_dealloc = Some(f),
+                _ => {}
+            }
+        }
+        let heap_alloc = heap_alloc.ok_or_else(|| format_err!("failed to find heap alloc"))?;
+        let heap_dealloc =
+            heap_dealloc.ok_or_else(|| format_err!("failed to find heap dealloc"))?;
+
+        // Create a shim function that looks like:
+        //
+        // (func __wbindgen_object_clone_ref (param i32) (result i32)
+        //      (local i32)
+        //      (table.set
+        //          (tee_local 1 (call $heap_alloc))
+        //          (table.get (local.get 0)))
+        //      (local.get 1))
+        let mut builder = walrus::FunctionBuilder::new();
+        let arg = module.locals.add(ValType::I32);
+        let local = module.locals.add(ValType::I32);
+
+        let alloc = builder.call(heap_alloc, Box::new([]));
+        let tee = builder.local_tee(local, alloc);
+        let get_arg = builder.local_get(arg);
+        let get_table = builder.table_get(table, get_arg);
+        let set_table = builder.table_set(table, tee, get_table);
+        let get_local = builder.local_get(local);
+
+        let ty = module.types.add(&[ValType::I32], &[ValType::I32]);
+        let clone_ref = builder.finish(ty, vec![arg], vec![set_table, get_local], module);
+        let name = "__wbindgen_object_clone_ref".to_string();
+        module.funcs.get_mut(clone_ref).name = Some(name);
```

Comment:
We should add a method for names to the builder. I also vaguely wanted this before.

Filed https://github.com/rustwasm/walrus/issues/61